From bb5454227188c13e1e14d71c1189fea754869cc5 Mon Sep 17 00:00:00 2001
From: Heitor Farhat <heitor.farhat@windriver.com>
Date: Fri, 20 Mar 2026 14:16:06 -0300
Subject: [PATCH] csi-powerscale: add label app.starlingx.io/component

This patch consists of changing the strategy from the Deployment
resource and add the app.starlingx.io/component label in all pods.
The strategy was needed changing because in the reapply the old
replicasets were not being deleted. The default value was rollingUpdate.
In this case, it doesn't affect functionality.

Signed-off-by: Heitor Farhat <heitor.farhat@windriver.com>
---
 charts/csi-isilon/templates/controller.yaml | 5 ++---
 charts/csi-isilon/templates/node.yaml       | 1 +
 charts/csi-isilon/values.yaml               | 3 +++
 3 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/charts/csi-isilon/templates/controller.yaml b/charts/csi-isilon/templates/controller.yaml
index 8c3bffb..b3e676b 100644
--- a/charts/csi-isilon/templates/controller.yaml
+++ b/charts/csi-isilon/templates/controller.yaml
@@ -172,15 +172,14 @@ spec:
   replicas: {{ required "Must provide the number of controller instances to create." .Values.controller.controllerCount }}
   {{- end }}
   strategy:
-    type: RollingUpdate
-    rollingUpdate:
-      maxUnavailable: 1
+    type: Recreate
   template:
     metadata:
       annotations:
         kubectl.kubernetes.io/default-container: "driver"
       labels:
         app: {{ .Release.Name }}-controller
+        app.starlingx.io/component: {{ ternary "application" "platform" .Values.labels.isApplication }}
     spec:
       serviceAccount: {{ .Release.Name }}-controller
       affinity:
diff --git a/charts/csi-isilon/templates/node.yaml b/charts/csi-isilon/templates/node.yaml
index 515c4bf..628019f 100644
--- a/charts/csi-isilon/templates/node.yaml
+++ b/charts/csi-isilon/templates/node.yaml
@@ -106,6 +106,7 @@ spec:
         kubectl.kubernetes.io/default-container: "driver"
       labels:
         app: {{ .Release.Name }}-node
+        app.starlingx.io/component: {{ ternary "application" "platform" .Values.labels.isApplication }}
 {{- if or .Values.podmon.enabled .Values.csm.resiliency }}
         driver.dellemc.com: dell-storage
 {{- end }}
diff --git a/charts/csi-isilon/values.yaml b/charts/csi-isilon/values.yaml
index 92029ab..2a8bcf9 100644
--- a/charts/csi-isilon/values.yaml
+++ b/charts/csi-isilon/values.yaml
@@ -457,3 +457,6 @@ storageClasses:
 snapshotClasses:

 proxyAuthzTokens:
+
+labels:
+  isApplication: false
--
2.34.1

